pkg/pillar/cmd/zedagent/localinfo.go

```diff
@@ -192,73 +197,135 @@ func processReceivedAppCommands(ctx *getconfigContext, cmdList *profile.LocalApp
 			continue
 		}
 		// Try to find the application instance.
-		ais := findAppInstance(ctx, appUUID, displayName)
-		if ais == nil {
+		appInst := findAppInstance(ctx, appUUID, displayName)
```

Comment:
This is the AppInstanceConfig received from zedagent, right?
It is confusing to modify that (the appInst.RestartCmd.Counter++ etc below) since now you are locally having something which is different than what was published (and it already confused me).

Reply:
Ah - this is in zedagent. Now I'm even more confused. zedagent is normally just protobuf to json conversion and here you have purge/restart logic. Can't that logic live in zedmanager?

Reply:
The only problem is the VolumeConfig, which goes directly from zedagent to volummgr. I would need to "re-route" it to go through zedmanger (to get local gen. counter). Do you see any potential problem this could introduce?

Reply:
We some factoring out of the current code I think having this in zedagent is fine for now.